Y/yaml_cpp/build_tarballs.jl

```diff
@@ -3,24 +3,32 @@
 using BinaryBuilder, Pkg
 
 name = "yaml_cpp"
-version = v"0.7.0"
+version = v"0.5.3"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/jbeder/yaml-cpp/archive/refs/tags/yaml-cpp-$(version).zip",
-                  "4d5e664a7fb2d7445fc548cc8c0e1aa7b1a496540eb382d137e2cc263e6d3ef5")
+    GitSource("https://github.com/jbeder/yaml-cpp.git", "b57efe94e7d445713c29f863adb8c23438eaa217"),
+    DirectorySource("bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/yaml-cpp*/
+
+patch -p1 < ../patches/boost_detail_iterator_obsolete.patch
+
+if [[ $target == *-apple-darwin* || $target == *-freebsd* ]]; then
+    cmake_extra_args=-DCMAKE_C_FLAGS=-D_LIBCPP_ENABLE_CXX17_REMOVED_FEATURES
+fi
+
 mkdir build && cd build
 cmake .. \
     -DCMAKE_INSTALL_PREFIX=${prefix} \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
     -DCMAKE_BUILD_TYPE=Release \
-    -DYAML_BUILD_SHARED_LIBS=ON \
-    -DYAML_CPP_BUILD_TESTS=OFF
+    -DBUILD_SHARED_LIBS=ON \
+    -DYAML_CPP_BUILD_TOOLS=OFF \
```

Comment:
Might be better to patch `CMakeLists.txt` to just disabled the tests - rather than also disabling build of the (parsing) tools...

Reply:
Tests can be disabled by removing https://github.com/jbeder/yaml-cpp/blob/b57efe94e7d445713c29f863adb8c23438eaa217/CMakeLists.txt#L331 , but building for Clang-target like `x86_64-apple-darwin`, or `x86_64-unknown-freebsd`, fails on building tools (despite having provided `-DCMAKE_C_FLAGS=-D_LIBCPP_ENABLE_CXX17_REMOVED_FEATURES` to CMake):
```
[ 91%] Building CXX object util/CMakeFiles/sandbox.dir/sandbox.cpp.o
cd /workspace/srcdir/yaml-cpp/build/util && /opt/bin/x86_64-unknown-freebsd13.2-libgfortran3-cxx11/x86_64-unknown-freebsd13.2-clang++ --sysroot=/opt/x86_64-unknown-freebsd13.2/x86_64-unknown-freebsd13.2/sys-root/  -I/workspace/srcdir/yaml-cpp/src -I/workspace/srcdir/yaml-cpp/include -O2 -MD -MT util/CMakeFiles/sandbox.dir/sandbox.cpp.o -MF CMakeFiles/sandbox.dir/sandbox.cpp.o.d -o CMakeFiles/sandbox.dir/sandbox.cpp.o -c /workspace/srcdir/yaml-cpp/util/sandbox.cpp
In file included from /workspace/srcdir/yaml-cpp/util/read.cpp:5:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:10:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:43:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Scanner> m_pScanner;
  ~~~~~^
In file included from /workspace/srcdir/yaml-cpp/util/sandbox.cpp:5:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:10:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:43:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Scanner> m_pScanner;
  ~~~~~^
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:44:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Directives> m_pDirectives;
  ~~~~~^
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:44:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Directives> m_pDirectives;
  ~~~~~^
In file included from /workspace/srcdir/yaml-cpp/util/sandbox.cpp:5:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:11:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/emitter.h:125:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<EmitterState> m_pState;
  ~~~~~^
In file included from /workspace/srcdir/yaml-cpp/util/read.cpp:5:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:11:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/emitter.h:125:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<EmitterState> m_pState;
  ~~~~~^
In file included from /workspace/srcdir/yaml-cpp/util/parse.cpp:6:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:10:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:43:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Scanner> m_pScanner;
  ~~~~~^
/workspace/srcdir/yaml-cpp/include/yaml-cpp/parser.h:44:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<Directives> m_pDirectives;
  ~~~~~^
In file included from /workspace/srcdir/yaml-cpp/util/parse.cpp:6:
In file included from /workspace/srcdir/yaml-cpp/include/yaml-cpp/yaml.h:11:
/workspace/srcdir/yaml-cpp/include/yaml-cpp/emitter.h:125:8: error: no template named 'auto_ptr' in namespace 'std'
  std::auto_ptr<EmitterState> m_pState;
  ~~~~~^
3 errors generated.
```

Reply:
Dropping further investigation into this